vortex-scan/src/range_scan.rs

```diff
@@ -18,7 +18,7 @@ pub struct RangeScanner {
 
 enum State {
     // First we run the filter expression over the full row range.
-    FilterEval((FilterMask, ExprRef)),
+    FilterEval((FilterMask, ExprRef, Vec<ExprRef>)),
```

Comment:
Might be time to pull this into a `struct FilterEval { mask, conjuncts: Vec<ExprRef> }`